app/controllers/incomes_controller.rb

```diff
@@ -0,0 +1,59 @@
+class IncomesController < ApplicationController
+  before_action :require_login
+  before_action :require_current_budget, only: [:show, :edit, :destroy]
+
+  def index
+    @incomes = current_budget.incomes.by_name
+  end
+
+  def new
+    @income = Income.new
```

Comment:
A suggestion to go through our `current_budget` here as well, i.e. `current_budget.incomes.new`